src/Microsoft.Extensions.Caching.Memory/CacheEntry.cs

```diff
@@ -172,12 +170,12 @@ public void Dispose()
             }
         }
 
-        internal bool CheckExpired(DateTimeOffset now)
+        public bool CheckExpired(DateTimeOffset now)
         {
             return _isExpired || CheckForExpiredTime(now) || CheckForExpiredTokens();
         }
 
-        internal void SetExpired(EvictionReason reason)
+        public void SetExpired(EvictionReason reason)
```

Comment:
It's a little odd for this to be public. The only EvictionReason that would make sense for the eviction algorithm to set would be Capacity. Expired is the next best reason, but CheckExpired should set that. Maybe make a new method that's just Evict and always uses Capacity as the reason?